ares/sfc/ppu/ppu.hpp

```diff
@@ -1,9 +1,50 @@
-#if defined(PROFILE_PERFORMANCE)
-#include "../ppu-performance/ppu.hpp"
-#else
-struct PPU : Thread, PPUcounter {
+struct PPUBase {
+  struct Implementation : Thread {
```

Comment:
Is there a reason we need a separate Implementation type below PPUBase? Couldn't we just have a PPU class that PPUperformance and PPUaccuracy inherit from?

Or is that too-high performance overhead?

Reply:
The ppu global would need to become a pointer, putting it at odds with every other component in the system.

Reply:
If there’s little-to-no performance impact, that may actually be a cleaner solution overall?

Reply:
PPUBase is an abstraction that hides the fact that there are two PPU implementations from other parts of the system that don't need to be aware of this detail. I'd say it makes the rest of the code cleaner, but mechanically speaking it would not be difficult to replace all the instances of "ppu." with "ppu->" and "ppu" with "*ppu". It would just make code that performs the same action on multiple components (like System::load and unload) slightly irregular.

Reply:
I'm probably just nitpicking, but whatever we do won't be as clean as how everything works with the compile-time PPUs, and there's not really a way to solve that; I do think we need runtime switching as an option, it just makes the code ugly.

But we may have to live with that.